src/libcmd/installables.cc

```diff
@@ -615,42 +618,118 @@ std::tuple<std::string, FlakeRef, InstallableValue::DerivationInfo> InstallableF
+    auto type = attr->maybeGetStringAttr(state->sType).value_or("");
 
-    auto drvPath = attr->forceDerivation();
+    if (type == "derivation") {
 
-    std::set<std::string> outputsToInstall;
-    std::optional<NixInt> priority;
+        auto drvPath = attr->forceDerivation();
 
-    if (auto aMeta = attr->maybeGetAttr(state->sMeta)) {
-        if (auto aOutputsToInstall = aMeta->maybeGetAttr("outputsToInstall"))
-            for (auto & s : aOutputsToInstall->getListOfStrings())
-                outputsToInstall.insert(s);
-        if (auto aPriority = aMeta->maybeGetAttr("priority"))
-            priority = aPriority->getInt();
-    }
+        std::set<std::string> outputsToInstall;
+        std::optional<NixInt> priority;
+
+        if (auto aMeta = attr->maybeGetAttr(state->sMeta)) {
+            if (auto aOutputsToInstall = aMeta->maybeGetAttr("outputsToInstall"))
+                for (auto & s : aOutputsToInstall->getListOfStrings())
+                    outputsToInstall.insert(s);
+            if (auto aPriority = aMeta->maybeGetAttr("priority"))
+                priority = aPriority->getInt();
+        }
 
-    if (outputsToInstall.empty() || std::get_if<AllOutputs>(&outputsSpec)) {
-        outputsToInstall.clear();
-        if (auto aOutputs = attr->maybeGetAttr(state->sOutputs))
-            for (auto & s : aOutputs->getListOfStrings())
-                outputsToInstall.insert(s);
+        if (outputsToInstall.empty() || std::get_if<AllOutputs>(&outputsSpec)) {
+            outputsToInstall.clear();
+            if (auto aOutputs = attr->maybeGetAttr(state->sOutputs))
+                for (auto & s : aOutputs->getListOfStrings())
+                    outputsToInstall.insert(s);
+        }
+
+        if (outputsToInstall.empty())
+            outputsToInstall.insert("out");
+
+        if (auto outputNames = std::get_if<OutputNames>(&outputsSpec))
+            outputsToInstall = *outputNames;
+
+        auto drvInfo = DerivationInfo {
+            .drvPath = std::move(drvPath),
+            .outputsToInstall = std::move(outputsToInstall),
+            .priority = priority,
+        };
+
+        return {attrPath, getLockedFlake()->flake.lockedRef, std::move(drvInfo)};
     }
 
-    if (outputsToInstall.empty())
-        outputsToInstall.insert("out");
+    else if (type == "configurable") {
 
-    if (auto outputNames = std::get_if<OutputNames>(&outputsSpec))
-        outputsToInstall = *outputNames;
+        // FIXME: use eval cache
 
-    auto drvInfo = DerivationInfo {
-        .drvPath = std::move(drvPath),
-        .outputsToInstall = std::move(outputsToInstall),
-        .priority = priority,
-    };
+        auto & build = attr->getAttr("build")->forceValue();
 
-    return {attrPath, getLockedFlake()->flake.lockedRef, std::move(drvInfo)};
+        struct ValueTree
+        {
+            std::map<std::string, std::variant<Value *, ValueTree>> children;
+        };
+
+        ValueTree tree;
+
+        if (cmd) {
```

Comment:
If we want other things besides drvs/installables to be configurable, we should probably extract everything from here to [line 713](https://github.com/NixOS/nix/pull/6583/files#diff-71050ce7f0123d7c142fef6cf61d196f17109ebd49dbcf41211650d154cf5484R713) into a separate function that can be used for other interfaces.